src/forms/createEqualTo.js

```diff
@@ -0,0 +1,52 @@
+import i18n from '@dhis2/d2-i18n'
+
+const isEmpty = value =>
+    typeof value === 'undefined' || value === null || value === ''
+
+export const requiredArgumentErrorMessage =
+    'Incorrect arguments provided when creating validator'
+
+export const requireArgument = (value, type) => {
+    if (isEmpty(value) || typeof value !== type) {
+        throw new Error(requiredArgumentErrorMessage)
+    }
+}
+
+const arrayIndexRegex = /\[(\d+)\]$/
+const delimiter = '.'
+const getByPath = (path, object) => {
+    const segments = path.split(delimiter)
+    const value = segments.reduce((curValue, segment) => {
+        if (!curValue) return curValue
+
+        if (segment.match(arrayIndexRegex)) {
+            const arraySegment = segment.replace(arrayIndexRegex, '')
+            const arrayIndex = parseInt(segment.match(arrayIndexRegex)[1], 10)
+
+            if (!curValue[arraySegment]) return curValue[arraySegment]
+            return curValue[arraySegment][arrayIndex]
+        }
+
+        const segmentValue = curValue[segment]
+        if (!segmentValue) return null
+        return segmentValue
+    }, object)
+
+    return value
+}
+
+export const createEqualTo = (key, description) => {
+    requireArgument(key, 'string')
+
+    const errorMessage = i18n.t(
+        'Please make sure the value of this input matches the value in "{{otherField}}".',
+        { otherField: description || key }
+    )
+
+    return (value, allValues) => {
+        const comparisonValue = getByPath(key, allValues)
+        return isEmpty(value) || value === comparisonValue
+            ? undefined
+            : errorMessage
+    }
+}
```

Comment:
Could well be that I am missing some nuances here, but I'm not sure you needed to craft these validators by hand. In `@dhis2/forms` we have an `isEmpty` and `createEqualTo` validator too. Did these have some issues? Or does this app need a unique implementation?

If the latter is the case, I think it'd be good to give these validators some distinguishing names (i.e. prevent naming overlap with the validators in @dhis2/forms).

Reply:
`isEmpty` is an internal implementation of the forms library, it's not exposed. I tried to use createEqualTo, but it doesn't work with paths (e. g. `specialChars[1].value`), which is why I added this one. I kept using the same names, as ideally, we port this to the prop types library in a follow up PR

Reply:
👍  makes perfect sense! TBH, I don't even think I realised we should keep the `paths` notation in mind when I wrote those validators.